mantidproject/mantid
# FindSXPeaks take goniometer angle into account for Q of found peaks

Description of work.

Peaks found with FindSXPeaks now take the input workspace's goniometer angle into account when calculating their QSample coordinates.

Note: This is based off of the branch for #20052, so make sure [that PR](https://github.com/mantidproject/mantid/pull/20104) is merged before merging this PR.

**To test:**

Fixes #20060.

1. Load a (raw) dataset that contains an `rrot` sample parameter. Extract the peaks and an MD Workspace in Q3D from it. Below is an example script using a run from ALF with the sample at 55°.
```python
alf = Load("ALF75142.raw")
alf_peaks = FindSXPeaks(alf, PeakFindingStrategy="AllPeaks", AbsoluteBackground=50, ResolutionStrategy="AbsoluteResolution", TofResolution=500, PhiResolution=2, TwoThetaResolution=2)
alf_md = ConvertToMD(InputWorkspace=alf, QDimensions="Q3D", dEAnalysisMode="Elastic", Q3DFrames="AutoSelect", MinValues="-10,-10,-10", MaxValues="10,10,10", SplitInto=100, MaxRecursionDepth=1, MinRecursionDepth=1)
```
If you view the MD workspace in the slice viewer and overlay the peaks workspace, the peak objects should ideally be aligned with the visible peaks in the data (depending on the input dataset you use, the quality of the peaks can vary) but at the very least, they should occupy the same region in Q space as the plot in the slice viewer.

2. Repeat the process, but this time, set the goniometer on the input workspace first.

```python
SetGoniometer(Workspace=alf, Axis0="rrot, 0, 1, 0, 1")
alf_peaks_rot = FindSXPeaks(alf, PeakFindingStrategy="AllPeaks", AbsoluteBackground=50, ResolutionStrategy="AbsoluteResolution", TofResolution=500, PhiResolution=2, TwoThetaResolution=2)
alf_md_rot = ConvertToMD(InputWorkspace=alf, QDimensions="Q3D", dEAnalysisMode="Elastic", Q3DFrames="AutoSelect", MinValues="-10,-10,-10", MaxValues="10,10,10", SplitInto=100, MaxRecursionDepth=1, MinRecursionDepth=1)
```
     
This gives you an MD workspace that taking the sample angle into account for its conversion to Q3D. If you overlay `alf_peaks` onto `alf_md_rot`, you should see those peaks now lying outside of the Q space occupied by the event data. If you overlay `alf_peaks_rot` however, you should see the same result as in step 1.


**Release Notes** 
Release notes added [here](https://github.com/mantidproject/mantid/commit/fec25e53d4b2e0068c237dd7e18bb19431b63786)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
